Check for working CXX compiler: /opt/rocm/hip/bin/hipcc - broken

This error occurs when building `rccl`, `miopen-hip` and `rocalution`.


```
==> Starting build()...
-- The CXX compiler identification is unknown
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: /opt/rocm/hip/bin/hipcc
-- Check for working CXX compiler: /opt/rocm/hip/bin/hipcc - broken
CMake Error at /usr/share/cmake-3.18/Modules/CMakeTestCXXCompiler.cmake:59 (message):
  The C++ compiler

    "/opt/rocm/hip/bin/hipcc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /build/rccl/src/build/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/usr/bin/make cmTC_74818/fast && /usr/bin/make  -f CMakeFiles/cmTC_74818.dir/build.make CMakeFiles/cmTC_74818.dir/build
    make[1]: Entering directory '/build/rccl/src/build/CMakeFiles/CMakeTmp'
    Building CXX object CMakeFiles/cmTC_74818.dir/testCXXCompiler.cxx.o
    /opt/rocm/hip/bin/hipcc   -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt  -o CMakeFiles/cmTC_74818.dir/testCXXCompiler.cxx.o -c /build/rccl/src/build/CMakeFiles/CMakeTmp/testCXXCompiler.cxx
    clang-12: error: cannot find HIP runtime. Provide its path via --rocm-path, or pass -nogpuinc to build without HIP runtime.
    clang-12: error: cannot find HIP runtime. Provide its path via --rocm-path, or pass -nogpuinc to build without HIP runtime.
    make[1]: *** [CMakeFiles/cmTC_74818.dir/build.make:85: CMakeFiles/cmTC_74818.dir/testCXXCompiler.cxx.o] Error 1
    make[1]: Leaving directory '/build/rccl/src/build/CMakeFiles/CMakeTmp'
    make: *** [Makefile:140: cmTC_74818/fast] Error 2
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:12 (project)


-- Configuring incomplete, errors occurred!
See also "/build/rccl/src/build/CMakeFiles/CMakeOutput.log".
See also "/build/rccl/src/build/CMakeFiles/CMakeError.log".
==> ERROR: A failure occurred in build().
```

--- a/hip-rocclr/.SRCINFO
+++ b/hip-rocclr/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = hip-rocclr
 	pkgdesc = Heterogeneous Interface for Portability ROCm
 	pkgver = 3.9.0
-	pkgrel = 1
+	pkgrel = 2
 	url = https://rocmdocs.amd.com/en/latest/Installation_Guide/HIP.html
 	arch = x86_64
 	license = MIT

--- a/hip-rocclr/PKGBUILD
+++ b/hip-rocclr/PKGBUILD
@@ -1,7 +1,7 @@
 # Maintainer: acxz <akashpatel2008 at yahoo dot com>
 pkgname=hip-rocclr
 pkgver=3.9.0
-pkgrel=1
+pkgrel=2
 pkgdesc="Heterogeneous Interface for Portability ROCm"
 arch=('x86_64')
 url='https://rocmdocs.amd.com/en/latest/Installation_Guide/HIP.html'
@@ -39,6 +39,10 @@ build() {
 
 package() {
   DESTDIR="$pkgdir" make -C build install
+
+  # clang from llvm-amdgpu may look for hipVersion in a different directory
+  install -d "$pkgdir/opt/rocm/bin"
+  ln -s '/opt/rocm/hip/bin/.hipVersion' "$pkgdir/opt/rocm/bin/.hipVersion"
 
   # add links (hipconfig is for rocblas with tensile)
   install -d "$pkgdir/usr/bin"

